frontend/src/ui/Select/Select.tsx

```diff
@@ -4,37 +4,83 @@ import Box from '@mui/joy/Box'
 import { styled } from '@mui/joy/styles'
 import { AutocompleteProps } from '@mui/joy/Autocomplete'
 
-type Option = {
-  value: string
+type SelectProps<T> = {
   label: string
-}
+} & AutocompleteProps<T, undefined, undefined, undefined>
 
-type SelectProps = {
+type FreeSoloProps = {
   label: string
-  value?: string
-} & Omit<AutocompleteProps<Option, undefined, undefined, undefined>, 'value'>
+} & AutocompleteProps<string, undefined, undefined, true>
+
+export const Select = <T,>({
+  options,
+  sx,
+  onChange,
+  name,
+  value,
+  label,
+  loading,
+  getOptionLabel,
+  disabled
+}: SelectProps<T>) => {
+  const selectButtonId = `select-button-${name}`
+  const selectLabelId = `select-label-${name}`
+
+  return (
+    <Box sx={{ position: 'relative', zIndex: 0 }}>
+      <StyledLabel htmlFor={selectButtonId} id={selectLabelId}>
+        {label}
+      </StyledLabel>
+      <Autocomplete
+        onChange={onChange}
+        name={name}
+        value={value}
+        sx={sx}
+        options={options}
+        autoSelect
+        autoHighlight
+        slotProps={{ input: { sx: { pt: '10px' } } }}
+        loading={loading}
+        getOptionLabel={getOptionLabel}
+        disabled={disabled}
+      />
+    </Box>
+  )
+}
 
-export const Select = ({ options, sx, onChange, name, value, label, loading }: SelectProps) => {
+export const FreeSoloSelect = ({
```

Comment:
I created this second `Select` because of the `freeSolo` prop (https://mui.com/joy-ui/react-autocomplete/#search-input), it adds the the possibility of writing thing that are outside of the select options, in this case it was the time select for the timer, it has options within a 15min interval but you can still add different times.

Because of the way that the Select is built, when you add the `freeSolo` prop, the type of the value becomes `T | string` instead of just `T`, so instead of having to check in the `onChange`  if the option is a string even when we know it will never be a string I decided to create a another similar component that only has string options for those cases.